dashboard-app/src/features/dashboard/ByProject/useAggregateDataByProject.ts

```diff
@@ -0,0 +1,73 @@
+import { DependencyList, useEffect, useState } from 'react';
+import { useBatchRequest } from '../../../lib/hooks';
+import { CommunityBusinesses } from '../../../lib/api';
+import {
+  logsToAggregatedData,
+  AggregatedData,
+  IdAndName
+} from '../dataManipulation/logsToAggregatedData';
+import { tableType } from '../dataManipulation/tableType';
+
+
+interface UseAggregatedDataParams {
+  from: Date;
+  to: Date;
+  updateOn?: DependencyList;
+}
+
+export default ({ from, to, updateOn = [] }: UseAggregatedDataParams) => {
+  const [aggregatedData, setAggregatedData] = useState<AggregatedData>();
+
+  const {
+    loading,
+    results: [logsData, projectsData, activitiesData],
+    error,
+  } = useBatchRequest({
+    requests: [
+      {
+        ...CommunityBusinesses.configs.getLogs,
+        params: { since: from, until: to },
+        transformResponse: [(res: any) => res.result],
+      },
+      {
+        ...CommunityBusinesses.configs.getVolunteerProjects,
+        transformResponse: [(res: any) => res.result.map(({ id, name }: IdAndName) => ({ id, name }))],
+      },
+      {
+        ...CommunityBusinesses.configs.getVolunteerActivities,
+        transformResponse: [(res: any) => res.result.map(({ id, name }: IdAndName) => ({ id, name }))],
+      },
+    ],
+    updateOn: [...updateOn, from, to],
+  });
+
+  useEffect(() => {
+    if (loading || error) {
```

Comment:
Is this a new pattern? Why's this happening?

Reply:
This isn't new, it's in all the other `useAggregateData...` hooks too. We don't want that effect to do anything if the data is either loading or there's been an error.